controlplane/kubeadm/internal/machine_collection.go

```diff
@@ -47,11 +47,12 @@ func NewFilterableMachineCollection(machines ...*clusterv1.Machine) FilterableMa
 
 // NewFilterableMachineCollectionFromMachineList creates a FilterableMachineCollection from the given MachineList
 func NewFilterableMachineCollectionFromMachineList(machineList *clusterv1.MachineList) FilterableMachineCollection {
+	if machineList == nil {
+		return nil
```

Comment:
Should this return a 0 length FilterableMachineCollection here instead of nil? Not sure if there are any downstream effects of doing so.

Reply:
Was this reported by a linter?

Reply:
yeah there's a NPE immediately below this if you pass in a nil MachineList

Reply:
Maybe `log.Fatal`? I feel like guarding against nil pointers is almost the wrong thing to do, it seems like a programmer error if we're getting a nil as input here.

Reply:
If before we were throwing a panic, we can do the same here given that we don't return any errors. That said, if we were to remove the pointer, the function should still work like an empty list?

Reply:
It might be better to add a variable for the initialization size for the `make` call below, or to initialize it as 0 length. I believe ass this stands today if one were to pass a nil pointer and they tried to use the FilterableMachineCollection returned by this function and later tried to use the `Insert` method, they would get a panic on Insert.

This should accomplish what looks like the original intent here:
```go
	var initLength int
	if machineList != nil {
		initLength = len(machineList.Items)
	}
	ss := make(FilterableMachineCollection, initLength)
```

Reply:
Turns out if we just remove the nil check that used to happen here the linter doesn't care anymore. This will result in a panic which I still think is the right behavior.